File: food-delivery-ruby/restaurant-service/models/restaurant.rb
require 'activerecord'

class Restaurant < ActiveRecord::Base
  has_many :menu_items, dependent: :destroy
  validates :name, presence: true
  validates :address, presence: true

  def as_json(options = {})
    {
      id: id,
      name: name,
      address: address,
      phone: phone,
      cuisine_type: cuisine_type,
      rating: rating,
      created_at: created_at,
      updated_at: updated_at,
      menu_items: options[:include_menu] ? menu_items.map(&:as_json) : nil
    }.compact
  end
end

File: food-delivery-dotnet/restaurant-service/Models/Restaurant.cs
using System.ComponentModel.DataAnnotations;

namespace RestaurantService.Models
{
    public class Restaurant
    {
        public int Id { get; set; }
        
        [Required]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;
        
        [Required]
        [StringLength(200)]
        public string Address { get; set; } = string.Empty;
        
        [StringLength(20)]
        public string Phone { get; set; } = string.Empty;
        
        [StringLength(100)]
        public string Cuisine { get; set; } = string.Empty;
        
        public decimal Rating { get; set; }
        
        public bool IsActive { get; set; } = true;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        
        public virtual ICollection<MenuItem> MenuItems { get; set; } = new List<MenuItem>();
    }
}